vendor-overwrites/csslint/parserlib.js

```diff
@@ -1919,10 +1930,22 @@ self.parserlib = (() => {
     function andand() {
       // andand = seq ( " && " seq)*
       const m = [seq()];
+      let reqPrev = !isOptional(m[0]);
+      const required = [reqPrev];
       while (reader.readMatch(' && ')) {
-        m.push(seq());
+        const item = seq();
+        const req = !isOptional(item);
+        // Matcher.many apparently can't handle optional items first
+        if (req && !reqPrev) {
+          m.unshift(item);
+          required.unshift(req);
+        } else {
+          m.push(item);
+          required.push(req);
+          reqPrev = req;
+        }
       }
-      return m.length === 1 ? m[0] : Matcher.andand.apply(Matcher, m);
+      return m.length === 1 ? m[0] : Matcher.many(required, ...m);
```

Comment:
I don't see the benefit. The code is arcane and requires deep involvement/debugging to modify it in a meaningful fashion either way.

Reply:
Also note, `&&` (andand) has `required=true` by definition, this is an inherent characteristic behavior of this mode, which should not be modifiable.

Reply:
Oi, there are two `andand` methods in Matcher :confused: 
I'll dig git blame and see why.

Reply:
Ah, it was always that way: one is inherited, another one is static.

Reply:
So yeah we can remove the `static andand` since it's unused but I'd like to keep it for the sake of completeness. I've removed some stuff Stylus didn't need but that was IIRC quite big, unlike the single-line method we're discussing.